cf/latest-snapshot/src/index.ts

```diff
@@ -7,31 +9,56 @@ function basename(path: String) {
 }
 
 // Directly fetch the data for the latest snapshot of a given chain (eg. calibnet or mainnet)
-async function get_latest(env: Env, chain: string): Promise<Response> {
+async function get_latest(req_headers: Headers, env: Env, chain: string): Promise<Response> {
 	const listed = await env.FOREST_ARCHIVE.list({ prefix: chain + "/latest/" });
 	let latest = listed.objects.at(-1);
 	if (latest == null) {
 		return new Response(`No latest snapshot found ${chain}`, {
 			status: 404,
 		});
 	} else {
-		// Should we support range queries?
-		const object = await env.FOREST_ARCHIVE.get(latest.key);
+		const object = await env.FOREST_ARCHIVE.get(latest.key, {
+			range: req_headers,
+			onlyIf: req_headers,
+		});
+
 		if (object === null) {
 			return new Response('No latest snapshot found', {
 				status: 404,
 			});
 		}
+
 		const headers = new Headers();
+
+		// If the client requested a range, then we need to set the Content-Range header.
+		// The `parseRange` returns an error code if the range is not satisfiable (or not specified),
+		// so we can handle the range request only if it returns an array.
+		let status = 200;
+		let range = parseRange(object.size, req_headers.get("range") || "");
```

Comment:
Why not use the range from `object.range`?

Reply:
The function accepts strings, and not `R2Range`.

Reply:
Unless you are referring to replacing `parseRange` with an `R2Range` altogether, which would bloat the code a bit, given how `R2Rrange` is implemented:
```
declare type R2Range =
  | {
      offset: number;
      length?: number;
    }
  | {
      offset?: number;
      length: number;
    }
  | {
      suffix: number;
    };
```

Reply:
Wouldn't it be simpler than what you have now? Anyhow, I'll leave it in your capable hands.

Reply:
Sadly, I don't think it will be simpler, and it would potentially introduce some corner cases (length larger than the total size, suffix larger than the total size...)